miri-script/src/commands.rs

```diff
@@ -446,10 +449,28 @@ impl Command {
         Ok(())
     }
 
-    fn test(bless: bool, flags: Vec<OsString>) -> Result<()> {
+    fn test(bless: bool, mut flags: Vec<OsString>) -> Result<()> {
         let mut e = MiriEnv::new()?;
+
+        // Extract the target from env vars or CLI args if it exists.
+        let target = get_target(&flags)?;
+        if let Some(target) = target.as_deref() {
+            // This must be set, it is what make it actually cross-interpret.
+            e.sh.set_var("MIRI_TEST_TARGET", target);
+
+            if has_flag(&flags, "target") {
+                // We must remove the flag if it came from the CLI because test driver has no idea
+                // what the target flag does.
+                let position = flags.iter().position(|flag| flag == "--target").unwrap();
+
+                // Remove both the key and the value
+                flags.remove(position);
+                flags.remove(position);
```

Comment:
Yeah sorry that's exactly what I said I didn't like... randomly mutating things in the middle of the CLI argument stream.

I think the better approach is to
- make `Test` [here](https://github.com/rust-lang/miri/blob/f399ed23678ff15831588272cc1343a5398ff7be/miri-script/src/main.rs#L31-L36) have an extra `target: Option<String>`
- Adjust the parsing [here](https://github.com/rust-lang/miri/blob/f399ed23678ff15831588272cc1343a5398ff7be/miri-script/src/main.rs#L149) to be more like `run` (i.e., with a loop) to extract leading `--bless` and `--target` flags